kn/fotos/entities.py

```diff
@@ -88,19 +101,50 @@ def id(self):
 
     description = son_property(('description',))
     visibility = son_property(('visibility',))
+    cached_visibility = son_property(('cachedVisibility',))
 
     def required_visibility(self, user):
         if user is None:
             return frozenset(('world',))
-        if user == '!system' or 'webcie' in user.cached_groups_names:
+        if 'webcie' in user.cached_groups_names:
             return frozenset(('leden', 'world', 'hidden'))
         if 'leden' in user.cached_groups_names:
             return frozenset(('leden', 'world'))
         return frozenset(('world',))
 
+    def update_cached_visibility(self, parent, save=True):
+        '''
+        Update the cachedVisibility property
+        '''
+        if self.cached_visibility is not None:
+            return False
+
+        if self.path is None:
+            # root
+            parent_cached_visibility = self.visibility
+        else:
+            parent_cached_visibility = parent.cached_visibility
+
+        visibilities = actual_visibility(parent_cached_visibility) & \
+                       actual_visibility(self.visibility)
+        order = ['world', 'leden', 'hidden']+self.visibility
+        if visibilities:
+            for v in order:
+                if v in visibilities:
+                    cached_visibility = [v]
```

Comment:
Als we ons toch gaan beperken tot een totaal geordend rijtje visibilities, is het misschien makkelijker simpelweg hiervoor getallen te gebruiken.  Dan is dit allemaal makkelijker en leesbaarder.